### ISSUE TITLE
If the "image" attribute is empty, the instance view fails to load

### ISSUE THREAD

**Issue by drwahl** [alvise72/DroidStack#14]
When clicking on the "Instances" button from the main page (after account is selected), the app fails to render anything with the error:

ServersActivity.AsyncTaskOSListServers.onPostExecute: Value  at image of type java.lang.String cannot be converted to JSONObject

I believe this is happening because the "image" field in my instances is null.  I also suspect that this bug is located around https://github.com/alvise72/DroidStack/blob/master/src/org/stackdroid/activities/ServersActivity.java#L1389

**Comment by alvise72** [comment]
Hi David ! would you like to contribute to the code starting with fixes like this ? and development in future (e.g. test and support new releases, Kilo, Mikata, v3 Keystone API, etc.) ?
I've very few time and it's difficult to follow the App in this period, but I'd like very much not to leave it die...

**Comment by drwahl** [comment]
My java skills are lacking at best, but I'd b willing to at least try patching this bug.

As far as testing/supporting new releases: there are parts that would be easy for me to test, since we have a couple of OpenStack deployments currently (Juno, Kilo, and soon Mitaka/head).  I can try to include testing for bits we currently aren't doing yet, but I can't make any promises as to when I might be able to get to them.

This is app is a great way to interact with OpenStack from Android, so I too don't want to see it die.  Hopefully I will be able to help breath some life into this project.

I will try to get a patch in this week for this bug.

**Comment by alvise72** [comment]
> but I'd b willing to at least try patching this bug

yes please!

If you give me your ID, I'll give it grant to commit into the DroidStack repo. I would prefere this way, than branching it and loosing synchronization soon in the future. What you think ?

**Comment by alvise72** [comment]
Hi David !
Have you had the chance to get a patch for this bug ?

**Comment by drwahl** [comment]
I have not.  My Java-fu is clearly not strong enough.  I poked around a bit, but wasn't able to actually fix the issue (back to the books, I guess).

**Comment by alvise72** [comment]
I've tried to create an instance, and then delete the image source in OpenStack. Then I've started DroidStack and I was able to list all the instances and also get info about the instance started from the not existing image.
Could you please describe your scenario ? and how to reproduce the error you saw ?

thanks.

**Comment by drwahl** [comment]
The "image" attribute on an instance will be empty if the image no longer exists in glance.  So, if you create a glance image, boot a VM from that image, then remove that original glance image, the VM will have an empty "image" metadata.  This was causing droidstack to throw an error about some list/array/etc being empty (which I don't have handy at the moment, but can get if you need it).

**Comment by alvise72** [comment]
Hi  David,
I'm unable to reproduce your problem. I start a VM from a certain image. Then I go back to the image list and I delete the source image of the VM.
I go back to the server list and I touch that VM; what I get about image info is simply "N/A".

Do you get that error again also with the last update (v30, "0.9 beta") ?

thanks,

    Alvise

**Comment by alvise72** [comment]
Hi David, to support my last sentence I'm sending you a video that shows that your problem seems not to be present as fas as I can try it: https://dl.dropboxusercontent.com/u/33282416/DroidStack%20Issue14.mp4 .

Have you got this problem once again ? Are you sure that the procedure is exactly the same as I did ?

**Comment by drwahl** [comment]
Yup, that's exactly the procedure.  Let me see if I can get some more details as to what is happening.  Also, here is a screenshot of the exact error I'm seeing:
http://oc.drwahl.me/index.php/s/bFdMdJiXtCm4G5y
